// baileys-mongodb/lib/Mongo/auth-utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.initAuthCreds = void 0;
const crypto_1 = require("./crypto");
const crypto_2 = require("crypto");
const uuid_1 = require("uuid");
const generateRegistrationId = () => {
    return Uint16Array.from((0, crypto_2.randomBytes)(2))[0] & 16383;
};
const initAuthCreds = () => {
    const identityKey = crypto_1.Curve.generateKeyPair();
    return {
        noiseKey: crypto_1.Curve.generateKeyPair(),
        pairingEphemeralKeyPair: crypto_1.Curve.generateKeyPair(),
        signedIdentityKey: identityKey,
        signedPreKey: (0, crypto_1.signedKeyPair)(identityKey, 1),
        registrationId: generateRegistrationId(),
        advSecretKey: (0, crypto_2.randomBytes)(32).toString("base64"),
        processedHistoryMessages: [],
        nextPreKeyId: 1,
        firstUnuploadedPreKeyId: 1,
        accountSyncCounter: 0,
        accountSettings: {
            unarchiveChats: false
        },
        // mobile creds
        deviceId: Buffer.from((0, uuid_1.v4)().replace(/-/g, ""), "hex").toString("base64url"),
        phoneId: (0, uuid_1.v4)(),
        identityId: (0, crypto_2.randomBytes)(20),
        registered: false,
        backupToken: (0, crypto_2.randomBytes)(20),
        registration: {},
        pairingCode: undefined,
        lastPropHash: undefined,
        routingInfo: undefined
    };
};
exports.initAuthCreds = initAuthCreds;
